src/algokit/core/deploy.py

```diff
@@ -0,0 +1,118 @@
+# 1. User can call algokit deploy to different networks
+# 2. By default algokit cli contains configs for testnet and mainnet
+# 3. User can overwrite them by creating a config file in the project root
+
+
+import logging
+import os
+import sys
+from pathlib import Path
+
+import click
+import httpx
+from dotenv import dotenv_values
+
+from algokit.core import proc
+from algokit.core.conf import get_algokit_config
+from algokit.core.constants import ALGOKIT_CONFIG, ALGORAND_NETWORKS, AlgorandNetworkConfiguration
+
+if sys.version_info >= (3, 11):
+    pass
+else:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+def get_genesis_network_name(deploy_config: dict) -> str | None:
+    """
+    Get the network name from the genesis block.
+    :param deploy_config: Deploy configuration.
+    :return: Network name.
+    """
+
+    algod_server = deploy_config.get("ALGOD_SERVER")
+
+    if not algod_server:
+        raise click.ClickException("Missing ALGOD_SERVER in deploy configuration.")
+
+    try:
+        genesis_response = httpx.get(f"{algod_server}/genesis")
+        genesis_response.raise_for_status()
+        return genesis_response.json()["network"] if genesis_response.status_code == 200 else None  # noqa: PLR2004
+    except httpx.HTTPError:
+        logger.warning(f"Failed to load network name from {algod_server} due to HTTP error.", exc_info=True)
+        return None
+
+
+def load_deploy_config(network: str, project_dir: Path) -> dict:
+    """
+    Load the deploy configuration for the given network.
+    :param network: Network name.
+    :param project_dir: Project directory path.
+    :return: Deploy configuration.
+    """
+
+    env_file_path = project_dir / f".env.{network}"
+    if not env_file_path.exists():
+        if network in ALGORAND_NETWORKS:
+            logger.info(f"No .env file found at {env_file_path}, loading default configuration for {network}.")
+            return dict(ALGORAND_NETWORKS.get(network, {}))
+
+        raise click.ClickException(f"No .env file found at {env_file_path}")
+
+    # Extract network configuration
+    network_config = dotenv_values(env_file_path, verbose=True, interpolate=True)
+
+    # Fill missing keys with empty strings
+    for required_key in list(AlgorandNetworkConfiguration.__annotations__.keys()):
+        if required_key not in network_config:
```

Comment:
is `INDEXER_SERVER` always required? doesn't that depend on the command?

Reply:
It was required in the explore command as dapflow expects them in the url params and i haven't tweaked this constant when moving this to constants.py to be reused in other folders and its partially required when doing init on a typed client if lookup is done on creator and name. But otherwise by default it should not be required for lookup on ids so i'm setting ALGOD_SERVER as the only required default parameter for now, lmk what you think @achidlow